substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/service/GcNotificationRequest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2024, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, 2024, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.genscavenge.service;
```

Comment:
Please move everything in this package to a sub package in the GC independent project `com.oracle.svm.core
com.oracle.svm.core`.

Reply:
I tried doing that originally, but it seems I cannot import from `com.oracle.svm.core.genscavenge` from inside `com.oracle.svm.core`.  Is there a way to work around this? 

The build error:
`error: package com.oracle.svm.core.genscavenge does not exist`

Reply:
You will need to split the implementation into a GC independent and a GC-specific part. This will also make it easier to implement this for all Native Image GCs in the future:
- GC independent parts in `com.oracle.svm.core`: notification thread, notification logic, and all the GC independent logic that accesses the memory pools. You will probably need introduce an interface and implementation that interface on the GC-specific side.
- GC specific parts in `com.oracle.svm.core.genscavenge`: essentially everything that would be different for a completely different GC implementation (e.g., the logic in `GenScavengeMemoryPoolMXBeans`, how many memory pools a GC uses, or how the memory pools are named)

Reply:
I put the files into `com.oracle.svm.core.notification` and  `com.oracle.svm.core.gc` (the naming could maybe be improved).

`com.oracle.svm.core.gc` contains the interface `MemoryPoolMXBeansProvider` for accessing GC-specific memory pool info. 

I also put `AbstractGarbageCollectorMXBean` inside `com.oracle.svm.core.gc`. Different GCs can override methods like `gcThreadCount()` and  `isIncremental()` so that the notification infrastructure knows how to handle notifications for different collectors.